src/utils/chat.ts

```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+import OpenAI from 'openai';
+import {
+  CONFIG_API_KEY,
+  CONFIG_BASE_URL,
+  CONFIG_IMPROVEMENT_CUSTOM_PROMPT,
+  CONFIG_MODEL,
+  DEFAULT_MODEL,
+} from '../constants';
+
+export async function AskQuestion(question: string, selection?: string) {
+  const config = await chrome.storage.local.get([
+    CONFIG_API_KEY,
+    CONFIG_BASE_URL,
+    CONFIG_MODEL,
+    CONFIG_IMPROVEMENT_CUSTOM_PROMPT,
+  ]);
+
+  if (!config[CONFIG_API_KEY]) return '';
+
+  const openai = new OpenAI({
+    apiKey: config[CONFIG_API_KEY],
+    baseURL: config[CONFIG_BASE_URL] || undefined,
+    dangerouslyAllowBrowser: true,
+  });
+
+  const completion = await openai.chat.completions.create({
+    messages: [
+      {
+        role: 'user',
+        content: buildAskPrompt(
+          question,
+          selection
+        ),
+      },
+    ],
+    model: config[CONFIG_MODEL] || DEFAULT_MODEL,
+  });
+
+  return completion.choices[0].message.content?.trim() ?? '';
+}
+
+function buildAskPrompt(question: string, selection?: string) {
+  if (selection) {
+    return (
```

Comment:
this prompt seems a bit unclear to me, would it be more intuitive to ask GPT to answer the question based on the selection?

Reply:
The issue is that the reply is often:

"Sure thing! In order to solve xxxx you write:"

Where the answer includes a human response and explanation instead of simply giving the answer

Reply:
Hence why the prompt is forced to only provide the output

Reply:
This part makes sense to me. I'm a bit confused about how the user is expected to use this feature. I think when a user selects some text, they may want to: 
1 rewrite the selection maybe with some specific instructions, or
2 ask a question regarding the selection
It looks like the prompt tries to do 1 but it also tries to answer a question.

Reply:
I aim to get an improvement/latex as a response rather than an explanation. When selecting text it aims to answer the question with the selection as context.
Other than the existing "improvement" button in this extension the user can ask more specific improvements such as "make this shorter", "change the color of the headings in this table to red",... 

When not selecting text it just tries to answer the question with a response that can be inserted into the paper.

So your 1. point is correct. Answering questions about a selection is not my aim with this PR.

Github copilot solves this ambiguity using commands such as /explain and /fix that indicate what you want to do with a selection. Future changes could incorporate something similar.

Reply:
... Or we could add a "improve" and "explain" button?